metricbeat/module/ceph/mgr_cluster_disk/_meta/data.json

```diff
@@ -0,0 +1,154 @@
+{
+    "@timestamp": "2020-02-12T12:21:29.890Z",
+    "@metadata": {
+        "beat": "metricbeat",
+        "type": "_doc",
+        "version": "8.0.0"
+    },
+    "ecs": {
+        "version": "1.4.0"
+    },
+    "ceph": {
+        "mgr_cluster_disk": {
+            "stats": {
+                "total_bytes": 10737418240,
+                "total_avail_bytes": 8342470656,
+                "total_used_bytes": 1321205760,
+                "total_used_raw_bytes": 2394947584,
+                "total_used_raw_ratio": 0.22304686903953552,
+                "num_osds": 1,
+                "num_per_pool_osds": 1
+            },
+            "pools": [
```

Comment:
> Adding @jsoriano as there is an issue with nested objects.
> 
> I'm afraid that this might be a blocker for light module, hence considering either rewriting this or using some magic with processors.

Yes, this kind of arrays of objects are unsupported, this question appears with some frequency, see [this thread](https://github.com/elastic/beats/issues/13137) for example that also shows this problem with the http module.

Something similar was also discussed for the Azure input, there we receive messages formatted as JSON objects containing lists of records, we wanted an event per record, and there was no way to do it using processors. It was finally supported by handling this in the underlying Kafka input (see https://github.com/elastic/beats/pull/13965). We could consider doing something similar for the http module.

The reason processors cannot be used for that, is that they are not able to generate multiple events from a single event, what is what we would need here.

For this specific data I think that ideally we should generate:
* An event for the `disk`, that contains the global stats and maybe a summarization of the pools (as number of pools).
* An event per `pool`, that also contains a field with the name of the `disk` for correlation.

A note about nested objects. We could make the effort of supporting them, and then we could store events like this one without [unexpected results](https://www.elastic.co/guide/en/elasticsearch/reference/7.6/array.html), but it would be an important effort with (in my opinion) little benefits as the information stored this way would be more difficult to query, and we could hardly take advantage of existing Kibana applications.

Reply:
After adjusting the PR there is no need for nested arrays.